File: single-language-analysis/dataset-files/java/TestSmell138.java
import org.junit.Test;
import static org.junit.Assert.*;

public class TestSmell138 {

    @Test
    public void testKMeansElkanIterAttribute() {
        KMeans km = new KMeans("elkan", 1).fit(X);
        assertEquals(1, km.nIter_);
    }
}

File: single-language-analysis/dataset-files/python/test_smell_138.py
# Original URL: https://github.com/ogrisel/scikit-learn/blob/b20998d4ec606558ae86cba7ff5a84aea094c397/sklearn/cluster/tests/test_k_means.py#L942-L946

def test_smell_138():
    # Regression test on bad n_iter_ value. Previous bug n_iter_ was one off
    # it's right value (#11340).
    km = KMeans(algorithm="elkan", max_iter=1).fit(X)
    assert km.n_iter_ == 1